### DISCUSSION TITLE
Week 3 Status Updates

### DISCUSSION THREAD

**Discussion by roryeiffe** [roryeiffe/Fresh-Results#20] [General]
Respond to this thread with everything you've worked on from 4/1/21-4/8/21.

What did you work on this week?

What sort of technical blockers are you facing?

What are you going to work on next week?

What sort of new technologies did you learn, if any?

Put any links to commits/other contributions here.

Feel free to add more questions as you think of them!

**Comment by roryeiffe** [comment]
# Rory Eiffe

## What did you work on this week?

This week, I worked on a color picker component. It takes a list of colors and instantiates circles where each circle is one of the colors that are passed in. Then, the user can click on a color to set it as the default color for the website. I also worked on figuring out how to connect the backend code that filters out spoiler words to the front-end. I wrote some rudimentary code that loops through elements on the page and replaces words like "death", "dies", etc. with "<spoiler word>".  Finally, I met with the UI team and we picked a general color scheme for the website. 

## What sort of technical blockers are you facing?

No technical blockers at this point in time.  

## What are you going to work on next week?

Next week, I am going to work with Abdul to try to put color blocks over sentences with spoilers. 

## What sort of new technologies did you learn, if any?

I learned how to interact with the dom to replace words using this article: https://medium.com/@divakarvenu/lets-create-a-simple-chrome-extension-to-interact-with-dom-7bed17a16f42. The gist of it is you loop through the different elements on the webpage, replace whatever words are deemed inappropriate and then alter the page with the new text. 

Put any links to commits/other contributions here.

https://github.com/roryeiffe/Spoiler-Block/commit/d7471c3b64de9007121cb998cba56078c4df10dc
https://github.com/roryeiffe/Spoiler-Block/commit/13780f6c6b934f26bb475a76669f7af30968eec5
https://github.com/roryeiffe/Spoiler-Block/commit/dbd57694ce87db060c1cd26c25d9e95f606fec76
https://github.com/roryeiffe/Spoiler-Block/commit/8e42f12e61c8cada196c7ec713d89aaa1d06bcfc
https://github.com/roryeiffe/Spoiler-Block/issues/10
https://github.com/roryeiffe/Spoiler-Block/issues/9
https://github.com/roryeiffe/Spoiler-Block/wiki/Contribution-Guidelines
https://github.com/roryeiffe/Spoiler-Block/commit/98515f4e9a9fb0ea73fccb7fab0d566d0347e1fa

**Comment by Sushiman33** [comment]
What did you work on this week?

- Added more words to the spoiler words list, the current count is now ~250 words. 
- Found a good way of turning a JSON file into a dictionary 

What sort of technical blockers are you facing?
- Still just learning more about JavaScript 

What are you going to work on next week?
- Gonna work with Ben to make the changes to the backend  to have any word in the dictionary replaced

**Comment by SerenaChen1** [comment]
# Serena Chen

## What did you work on this week?

On week 3, I worked on fixing the bugs I was having with the more info icon. On hover, the tool tip text would show up, but it would be cut off at the edge of the extension popup, so I moved it to the right side of the icon. There was also an issue with the z-index of the tool tip text because it would show up underneath the other components (i.e., Threshold and ColorPicker), which I fixed as well.

## What sort of technical blockers are you facing?

I'm not facing any technical blockers currently.

## What are you going to work on next week?

For next week, I was given the task of creating a style guide. 

## What sort of new technologies did you learn, if any?

I didn't learn any new technologies this week, but I got the chance to play around with a lot of CSS, which I didn't have much experience with before. 

Put any links to commits/other contributions here.

https://github.com/roryeiffe/Spoiler-Block/issues/15
https://github.com/roryeiffe/Spoiler-Block/pull/18
de94c61c9f140a5d7d32c26d0a68db8bf10511ce

**Comment by VictoriaSavage526** [comment]
What did you work on this week?
I attempted to create a popup that displays to the user when they open a webpage to tell them the spoiler-y content on the page

What sort of technical blockers are you facing?
I am having issues creating the popup on my system and opened an issue

What are you going to work on next week?
I am going to work on trying to remedy the issue with my system and the extension on my system.

What sort of new technologies did you learn, if any?
I learned more react and JS this week.

**Comment by TheCovenant** [comment]
### What did you work on this week:

This week I worked on creating a component that on our extension's pop-up that enables them to create a 'keyword group'. When this group is created the user can add keywords that he/she believes indicates a spoiler is near. This allows the user to block keywords that are specific to shows that they are watching.

### What sort of technical blockers are you facing:

I'm not facing any significant technical blockers at this point. 

### What are you going to work on next week:

As I did not finish the component, I will continue to work on it the following week.

### What sort of new technologies did you learn, if any:

None, but I did solidify my React knowledge a bit more.

### Links to Contributions:
https://github.com/roryeiffe/Spoiler-Block/pull/39/commits/d5f23e865b6d4610e829c650673b566901d5289f
https://github.com/roryeiffe/Spoiler-Block/pull/39/commits/24ab7d1b2fa753b54a512828cdf4916f9885a5cb
